lib/awful/client.lua

```diff
@@ -1142,9 +1140,50 @@ capi.client.connect_signal("manage", function (c)
 end)
 capi.client.connect_signal("unmanage", client.focus.history.delete)
 
-
 capi.client.connect_signal("unmanage", client.floating.delete)
 
+-- Connect to "focus" signal, and allow to disable tracking.
+do
+    local disabled_count = 1
+    --- Disable history tracking.
+    --
+    -- See `awful.client.focus.history.enable_tracking` to enable it again.
+    -- @treturn int The internal value of `disabled_count` (calls to this
+    --   function without calling `awful.client.focus.history.enable_tracking`).
+    -- @function awful.client.focus.history.disable_tracking
+    function client.focus.history.disable_tracking()
+        disabled_count = disabled_count + 1
+        if disabled_count == 1 then
+            capi.client.disconnect_signal("focus", client.focus.history.add)
+        end
+        return disabled_count
+    end
+
+    --- Enable history tracking.
+    --
+    -- This is the default, but can be disabled
+    -- through `awful.client.focus.history.disable_tracking`.
+    -- @treturn boolean True if history tracking has been enabled.
+    -- @function awful.client.focus.history.enable_tracking
+    function client.focus.history.enable_tracking()
+        assert(disabled_count > 0)
+        disabled_count = disabled_count - 1
+        if disabled_count == 0 then
+            capi.client.connect_signal("focus", client.focus.history.add)
+        end
+        return disabled_count == 0
+    end
+
+    --- Is history tracking enabled?
+    -- @treturn[0] true True is history tracking is enabled.
+    -- @treturn[1] int The number of times that tracking has been disabled.
+    -- @function awful.client.focus.history.is_enabled
+    function client.focus.history.is_enabled()
+        return disabled_count == 0 or disabled_count
```

Comment:
No.. it should have been either `true` (`disabled_count` == 0), or `disabled_count`.
What you mean is returning two values `bool` and the number, right?
Makes more sense and seems to be less magical that way - amended.